spinflip/spin_flip_tools.py

```diff
@@ -121,14 +130,76 @@ def __init__(self, gradient_filename, merger_data_loc):
         #need matter part gradients from changing Y_e, n_b
         self.grad_S_R_mat, self.grad_S_L_mat = np.zeros(np.shape(self.grad_S_R_nu)), np.zeros(np.shape(self.grad_S_L_nu))
             
-        #total Sigma Gradients
+        #total Sigma Gradients (spacetime up, spacetime (gradient) low, x,y,z, F, F
         self.grad_S_R, self.grad_S_L = self.grad_S_R_nu + self.grad_S_R_mat, self.grad_S_L_nu + self.grad_S_L_mat
 
+    #total H_L and H_R gradients (ignoring extra terms in inverse p/abs)
+    #(spacetime low, x,y,z, F, F)
+    def grad_H_L(self, theta, phi):
+        basis = Basis(theta,phi)
+        return basis.kappa(self.grad_S_L)
+    def grad_H_R(self, theta, phi):
+        basis = Basis(theta,phi)
+        return basis.kappa(self.grad_S_R)
+   
+        
+    # plot of magnitude of gradient along minimizing resonant direction
```

Comment:
I think this comment goes along with the next function

Reply:
at each cell, if there is resonance, it for loops over azimuthal angles and finds the resonant polar angles at each azimuthal, then finds the gradients in those directions. the last step is min_gradients[x,y,z] = np.min (gradients) where the minimization happens. Unless I'm doing something wrong here

Reply:
No, I think I just misread the function. Thanks for helping me understand.